doc/SAIP4Extensions/SAI_P4EXT_API_Proposal.md

```diff
@@ -0,0 +1,279 @@
+SAI P4EXT API Proposal
+=====================
+
+Title       | SAI IPsec API Proposal
+------------|----------------------
+Authors     | Intel.
+Status      | Draft
+Type        | Standards track
+Created     | 03/18/2021
+SAI-Version | ?
+-------------------------------
+
+# Scope #
+
+This document defines the technical specifications for the API used to support P4 Extensions in Open Compute Project Switch Abstraction Interface (SAI). P4 Extensions can be seen as an abstraction over niche/device specific capabilities. It facilitates programming of these new features via the familiar SAI API model using a single P4 Extension Object. The exact feature enabled via the P4 Extensions itself is outside the scope of this document and SAI P4 Extensions API.
+
+# Overview #
+
+SAI P4 extension introduces a single new SAI object SAI_OBJECT_TYPE_P4EXT_ENTRY of object_type_oid. This object along with its attributes provides an abstraction of target specific configurable features otherwise not described in SAI via the P4 match action paradigm in a feature agnostic manner. In contrast to other SAI objects which are mapped 1:1 to an entity in the SAI Pipeline Netwokring Object Model, for eg. VLAN, Neighbor, Mirror etc.; SAI P4 Extensions introduces 1:N mapping between the  SAI_OBJECT_TYPE_P4EXT_ENTRY object and niche features Feature1, Feature2, etc. This new paradigm enables the
```

Comment:
We should explain how the P4 language enables describing the features and then segue into explaining the attributes and how they map to the P4 language.

Reply:
what is object_type_oid?

Reply:
Networking typo

Reply:
> what is object_type_oid?

SAI objects can be key based or oid based [sai_metadata_is_object_type_oid(object_type)](https://github.com/barefootnetworks/SAI/blob/master/inc/saiobject.h#L48-L55)